command/agent/agent.go

```diff
@@ -487,49 +545,66 @@ func (a *Agent) Stats() map[string]map[string]string {
 	return stats
 }
 
-func (a *Agent) createConsulConfig() {
-	cfg := &consul.ConsulConfig{
-		Addr:      a.config.ConsulConfig.Addr,
-		Token:     a.config.ConsulConfig.Token,
-		Auth:      a.config.ConsulConfig.Auth,
-		EnableSSL: a.config.ConsulConfig.EnableSSL,
-		VerifySSL: a.config.ConsulConfig.VerifySSL,
-		CAFile:    a.config.ConsulConfig.CAFile,
-		CertFile:  a.config.ConsulConfig.CertFile,
-		KeyFile:   a.config.ConsulConfig.KeyFile,
-	}
-	a.consulConfig = cfg
+// setupConsulSyncer creates the Consul task used by this Nomad Agent when
+// running in either Client and Server mode.
+func (a *Agent) setupConsulSyncer(shutdownCh chan struct{}) (err error) {
+	a.consulSyncer, err = consul.NewSyncer(a.config.Consul, a.logger)
+
+	return nil
 }
 
-// syncAgentServicesWithConsul syncs the client and server services with Consul
-func (a *Agent) syncAgentServicesWithConsul(clientHttpAddr string, serverHttpAddr string) error {
-	cs, err := consul.NewConsulService(a.consulConfig, a.logger)
-	if err != nil {
-		return err
-	}
-	a.consulService = cs
+// syncAgentServicesWithConsul syncs this Nomad Agent's services with Consul
+// when running in either Client or Server mode.
+func (a *Agent) syncAgentServicesWithConsul() error {
 	var services []*structs.Service
-	if a.client != nil && a.config.ConsulConfig.ClientServiceName != "" {
-		if err != nil {
-			return err
-		}
-		clientService := &structs.Service{
-			Name:      a.config.ConsulConfig.ClientServiceName,
-			PortLabel: clientHttpAddr,
+	if a.client != nil {
+		if a.config.Consul.ClientServiceName != "" {
```

Comment:
! I've put all of this logic behind `consul.auto_register`, but just realized this acts as a super convenient - albeit semi-obtuse - way to disable "`consul.auto_bootstrap`" without creating a new tunable.